flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -170,8 +170,12 @@ public ScopeFormats getScopeFormats() {
 	 * @param group       the group that contains the metric
 	 */
 	public void register(Metric metric, String metricName, AbstractMetricGroup group) {
-		if (reporter != null) {
-			reporter.notifyOfAddedMetric(metric, metricName, group);
+		try {
+			if (reporter != null) {
+				reporter.notifyOfAddedMetric(metric, metricName, group);
+			}
+		} catch (Exception e) {
+			LOG.error("Error while registering metric.", e);
```

Comment:
Why do we add the try-catch here? `notifyOfAddedMetric` does not declare thrown exceptions. Thus we try to catch unchecked exceptions here? Which unchecked exceptions did you encounter here?